src/named_goal.cpp

```diff
@@ -0,0 +1,77 @@
+disclaimer.
+
+2. Redistributions in binary form must reproduce the above copyright notice, this list of conditions and the following
+disclaimer in the documentation and/or other materials provided with the distribution.
+
+3. Neither the name of the copyright holder nor the names of its contributors may be used to endorse or promote products
+derived from this software without specific prior written permission.
+
+THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES,
+INCLUDING, BUT NOT LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
+DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL,
+SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR
+SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY,
+WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF
+THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+*/
+
+/* Author: Karl Kruusamäe */
+/* E-mail: karl.kruusamae@ut.ee */
+
+#include <moveit/move_group_interface/move_group_interface.h>
+
+int main(int argc, char** argv)
+{
+  // Basic ROS setup
+  // ^^^^^^^^^^^^^^^
+  ros::init(argc, argv, "pose_goal");
+  ros::NodeHandle node_handle;
+  ros::AsyncSpinner spinner(1);
+  spinner.start();
+
+  // MoveGroupInterface API can be found at:
+  // http://docs.ros.org/kinetic/api/moveit_ros_planning_interface/html/classmoveit_1_1planning__interface_1_1MoveGroupInterface.html
+
+  // MoveIt setup
+  // ^^^^^^^^^^^^
+  // A MoveGroupInterface instance (a client for the MoveGroup action) can be easily setup using just the name of the
+  // planning group you would like to control and plan for.
+  moveit::planning_interface::MoveGroupInterface move_group("xarm7");
+
+  // Planning to a predefined named pose
+  // ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+  // We can plan a motion for this group by specifying a target pose from the list of defined poses. The named poses can
+  // be defined for a particular planning group using the MoveIt Setup Assistant.
+
+  // Setting a named pose as target
+  move_group.setNamedTarget("home_pose");
+
+  // Calling the planner to compute the motion plan, which is then stored in my_plan.
+  // Note that we are just planning, not asking MoveGroupInterface to actually move the robot.
+  moveit::planning_interface::MoveGroupInterface::Plan my_plan;
+  moveit::planning_interface::MoveItErrorCode success = move_group.plan(my_plan);
+  if (success)
+  {
+    ROS_INFO("[movegroup_interface_demo/named_goal] Planning OK. Proceeding.");
+  }
+  else
+  {
+    ROS_WARN("[movegroup_interface_demo/named_goal] Planning failed. Shutting Down.");
+    ros::shutdown();
```

Comment:
We should make sure the "execute plan" part is not reached in case of a planning failure... either move the execution part into the if(success) block above or simply
```suggestion
    ros::shutdown();
    return 0;
```